modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -57,16 +57,17 @@ package object engine {
     */
   def switch(q: EventQueue)(id: Sequence.Id)(st: SequenceState): Handle[Unit] =
     resources.flatMap(
-      ores => modifyS(id)(
-        seq =>
-        if (st === SequenceState.Running)
-          // No resources being used by other running sequences
-          if (seq.toSequence.resources.intersect(ores).isEmpty)
-            Sequence.State.status.set(seq, st)
-          // Some resources are being used, sequence unchanged
-          else seq
-        else Sequence.State.status.set(seq, st)
-      )
+      ores => getS(id).flatMap {
```

Comment:
It comes from *other resource* but it's a very short lived value so I think it's not that important to convey any meaning, it could have just been called *x*.